### PR TITLE
Add helper and assets to run accesslint.js from a Rails app

### PR DESCRIPTION
**Pull request by ckundo**
- precompile accesslint.js.
- add a helper to load js into a layout
- update docs and tests
- update dummy app

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,3 +1,13 @@
...
+
+gem "pry"
+
+group :development, :test do
+  gem "rspec-rails"
+end
+
+group :test do
+  gem "capybara-webkit"
+end
```
Comment (geoffharcourt) [inline Gemfile:11]: Should all of these be in the `.gemspec` as development dependencies?

File: Gemfile [hunk 2]
```diff
@@ -1,3 +1,13 @@
 source 'https://rubygems.org'
 
 gemspec
+
+gem "pry"
```
Comment (geoffharcourt) [inline Gemfile]: Should this be in the `:development` group?

File: README.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
+# AccessLint::Rails
+
+AccessLint::Rails tests and reports accessibility errors in your Rails app.
+
+To run accessibility tests in the browser, include the linter JavaScript which
+will `POST` results to your application at a mounted endpoint.
+
+The endpoint raises an exception that will be captured and reported by your
+exception notifier.
```
Comment (geoffharcourt) [inline README.md]: Is the this still true? Doesn't it post to the AccessLint service now?

File: app/helpers/access_lint_helper.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
+module AccessLintHelper
+  def include_access_lint
+    if Rails.env.production?
+      js_file = "//accesslint.com/1.0/accesslint.js"
```
Comment (geoffharcourt) [inline app/helpers/access_lint_helper.rb]: Should we serve this from the same place we're reporting errors to? Or a CDN?

Maybe that's a future change, but moving it to the service app would let us do clever user-specific dynamic things in the JavaScript, e.g. some of the sampling stuff we were talking about.

File: app/helpers/access_lint_helper.rb [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    else
+      js_file = "accesslint"
+    end
+
+    javascript_include_tag(js_file) +
+    javascript_tag(<<-JS.strip_heredoc)
+      window.addEventListener("load", function load(){
+        window.removeEventListener("load", load, false);
```
Comment (geoffharcourt) [inline app/helpers/access_lint_helper.rb]: If there are other `load` listeners, will this cause problems for them?

File: Gemfile (hunk 2 above)
Comment (ckundo) [inline Gemfile]: Moved it to the gemspec.

File: Gemfile (hunk 1 above)
Comment (ckundo) [inline Gemfile:11]: Moving these breaks the spec in the dummy app.

File: app/helpers/access_lint_helper.rb (hunk 5 above)
Comment (geoffharcourt) [inline app/helpers/access_lint_helper.rb]: Oh, I see now. The function's name is `load`.

Comment (geoffharcourt) [inline app/helpers/access_lint_helper.rb]: Nitpick, but is it dangerous to have this with a global name?

Comment (ckundo) [inline app/helpers/access_lint_helper.rb]: @geoffharcourt this code was inspired by:

https://developer.mozilla.org/en-US/Add-ons/Code_snippets/On_page_load#Attaching_a_script

Also, `load` is properly scoped, and not exposed on `window`. The call to `removeEventListener` accepts the `load` function as an argument and will only remove it. 

This approach is preferred over setting `window.onload` for the sake of not clobbering existing listeners.

> -- @seanpdoyle

**Pull request merged**